File: frontend/src/services/billingService.js
import axiosInstance from "../utils/axiosInstance";

const billingService = {

    getBillingSummary: async () => {
        const response = await axiosInstance.get("/billing/summary");
        return response.data;
    },

    getAllBilling: async () => {
        const response = await axiosInstance.get("/billing/all");
        return response.data;
    },

    getDepartmentWiseCost: async () => {
        const response = await axiosInstance.get("/billing/department");
        return response.data;
    },

    markAsPaid: async (billingId) => {
        const response = await axiosInstance.put(`/billing/${billingId}/pay`);
        return response.data;
    },

    // Student / Faculty
    getMyBilling: async () => {
        const response = await axiosInstance.get("/billing/my");
        return response.data;
    },

    // Department Head
    getDepartmentBilling: async (userId) => {
        const response = await axiosInstance.get(
            `/billing/department-billing/${userId}`
        );
        return response.data;
    },

    // Institute Admin
    getInstitutionBilling: async (institutionId) => {
        const response = await axiosInstance.get(
            `/billing/institution/${institutionId}`
        );
        return response.data;
    }

};

export default billingService;

File: backend/src/main/java/com/labresource/backend/service/BillingService.java
package com.labresource.backend.service;

import com.labresource.backend.dto.BillingResponse;
import com.labresource.backend.dto.BillingSummaryResponse;
import com.labresource.backend.dto.DepartmentCostResponse;
import com.labresource.backend.entity.Billing;
import org.springframework.security.core.Authentication;
import java.util.List;

public interface BillingService {

    BillingSummaryResponse getBillingSummary();

    List<BillingResponse> getAllBilling();

    List<DepartmentCostResponse> getDepartmentWiseCost();

    Billing markAsPaid(Long billingId);
    // Student / Faculty
    List<BillingResponse> getMyBilling(Authentication authentication);
    // Department Head
    List<BillingResponse> getDepartmentBilling(Long userId);

    // Institute Admin
    List<BillingResponse> getInstitutionBilling(Long institutionId);

}